### ISSUE TITLE
Feature Request: Command Line Streamline

### ISSUE THREAD

**Issue by EasyG0ing1** [imothee/tmpdisk#40]
I see in the readme how to use the open command to create a ram drive from the command line, but once the drive is created, TMPDisk remains running in the background.

Could we add a command line switch for closing the program once the drive is created?

Also, are all of the command line arguments documented anywhere? Or are the command line arguments in the readme the only ones available?

Also, it would be nice to be able to specify the size units on the command line like:

```Bash
open -a /Applications/TmpDisk.app --args -name=TestDisk -size=64MB
open -a /Applications/TmpDisk.app --args -name=TestDisk -size=24GB
```
Or maybe
```Bash
open -a /Applications/TmpDisk.app --args -name=TestDisk -size=64 -units=MB
open -a /Applications/TmpDisk.app --args -name=TestDisk -size=24 -units=GB
```

**Comment by imothee** [comment]
Hi @EasyG0ing1 

I had only really added the cmd line prompt to make it easier to launch tmpdisk to manage the disks on startup (a feature request) but that was quickly superseded by the AutoCreate manager and run on login and pretty much just a legacy.

If your use case is just creating the TmpFS or Ram disks you can do it manually or create a simple bash shell around them instead of using tmpdisk since the extra features TmpDisk offers are only valid if TmpDisk is running.

```
sudo mount_tmpfs -s128M ~/.tmpfs/temp-128-mb-tmpfs
```

Or here's how to script up each of the variables for a plain ram disk

```swift
func createRamDiskTask(volume: TmpDiskVolume) -> String {
        let dSize = UInt64(volume.size) * 2048
        
        let filesystem: String = {
            switch (volume.caseSensitive, volume.journaled) {
            case (false, false):
                return "HFS+" // Mac OS Extended
            case (true, false):
                return "HFSX" // Mac OS Extended (Case-sensitive)
            case (false, true):
                return "JHFS+" // Mac OS Extended (Journaled)
            case (true, true):
                return "JHFSX" // Mac OS Extended (Case-sensitive, Journaled)
            }
        }()
        
        if volume.hidden {
            return "d=$(hdiutil attach -nomount ram://\(dSize)) && diskutil eraseDisk \(filesystem) %noformat% $d && newfs_hfs -v \"\(volume.name)\" \"$(echo $d | tr -d ' ')s1\" && hdiutil attach -nomount $d && hdiutil attach -nobrowse \"$(echo $d | tr -d ' ')s1\""
        } else {
            return "diskutil eraseVolume \(filesystem) \"\(volume.name)\" `hdiutil attach -nomount ram://\(dSize)`"
        }
    }
```

**Comment by imothee** [comment]
I'm going to overhaul this as part of 2.2.0 alongside the other bug with cli args.

**Comment by imothee** [comment]
Would love to get feedback on what's missing or unclear

```
TmpDisk Command Help
===================

Commands:
  help                    Display this help message

Parameters:
  name=VALUE              Set the disk name (default: TmpDisk)
  size=VALUE[MB|GB]       Set the disk size (default: 64MB)
  units=[MB|GB]           Set the size units (default: MB)
  fs=FILESYSTEM           Set the filesystem type (default: APFS)
                          Available: APFS, APFSX, HFS+, TMPFS, HFSX, JHFS+, JHFSX

Examples:
  tmpdisk name=MyDisk size=1GB
  tmpdisk size=512 fs=HFS+
  tmpdisk help

Note: Parameters can be prefixed with '-' (e.g., -name=MyDisk)
```

**Comment by EasyG0ing1** [comment]
@imothee 

Hi, just wanted to give you some feedback on the CLI (which will be awesome once it's dialed in) ...

After installing the CLI into ~/.local/bin here is what happens when I try to engage it using various options:

First of all, every time I run the `tmpdisk` command, it launches a "system tray" icon at the top of my screen and the program never exits out of the bash command, meaning that it hangs the bash shell that I run it from. What would be ideal, would be if the bash command never launched the GUI portion of the program at all, and then it simply creates the ram volume then exits out - potentially with a return code (0 meaning all is well) this way, it could be called from other programs or scripts where one could then continue doing other things with the ram drive where the use of the command doesn't hang the shell.

Second, if the tray icon exists and I run `tmpdisk` from bash, it simply returns me to a prompt and does nothing. I have to first quit the program by clicking on the tray icon and choosing Quit before running the bash command. This could be a problem when trying to use the bash command from scripts or other programs.

In addition to that, here are the results of various usages of the command:

```BASH
tmpdisk -name=RAMDisk -size=512 -fs=APFS
```

This creates the `RAMDisk` volume sized at 512 megabytes.

However:
```BASH
tmpdisk -name=RAMDisk -size=8GB -fs=APFS
tmpdisk -name=RAMDisk -size=8 -units=GB -fs=APFS
```

Both of these do the same thing: they launch the GUI system tray icon and also hang the bash shell, but neither creates a RAM volume of any kind.  In fact, I've not been able to get the CLI program to create a ram disk of any kind in the gigabyte range unless I don't specify the units and just use raw MB values ... for example:

```BASH
tmpdisk -name=RAMDisk -size=8192 -fs=APFS
```

Gives me an 8 gigabyte RAM drive, but that's the only way I can make an 8 gigabyte RAM drive. And of course, the bash shell remains blocked, and the system tray icon is also running ... ideally, neither of those behaviors would happen.

Finally, when I run `Check For Updates` from the system tray icon, I get this:

<img src="https://i.imgur.com/vmtPRTf.png" alt="screenshot" WIDTH="300"/>

I did these tests with version 2.2.1

**Comment by imothee** [comment]
Hi @EasyG0ing1 right now the cli is just a pathway into calling the main app logic. I didn't want to duplicate all the creation and management functionality into the standalone binary but I can definitely look to see if that's possible.

I fixed the update error, I'll take a look at why the cli is hanging on the other calls.

**Comment by imothee** [comment]
Totally overhauled as part of 2.3 beta https://github.com/imothee/tmpdisk/releases/tag/v2.3.0-beta
Would love some help testing

**Comment by EasyG0ing1** [comment]
@imothee I've written JavaFX apps in the past where the core functionality existed in isolated classes from the GUI. This way, I could implement command-line capability without ever launching the GUI portion of the code. So you just end up with two different interfaces to the same back-end functionality. Though I know refactoring code can sometimes be more grueling than a rewrite. Just wanted to throw that idea out there.

**Comment by imothee** [comment]
@EasyG0ing1 that's essentially what I did on the 2.3.0 beta release :) Was blocked on getting it signed by Apple but the newest version is ready for testing before release.

**Comment by EasyG0ing1** [comment]
@imothee  I'm curious as to why Apple wouldn't sign it.

**Comment by EasyG0ing1** [comment]
@imothee Also one suggestion: Implementing the ability to get the version from the CLI 

```bash
tmpdsk version
tmpdsk --version
tmpdsk -v
```

Something like that if possible.

So far so good on the CLI. No issues with the GUI at all. Nice Work! Thanks for taking the time to do that.

If you have some specific features that youd like me to test from a Java execution perspective, Id be happy to do that. Im going to test mainly the stuff I would use and see how it behaves in a JavaProcess call.

**Comment by imothee** [comment]
> [@imothee](https://github.com/imothee) I'm curious as to why Apple wouldn't sign it.

I decided to transfer the apple developer account from personal to my LLC and that process took almost 2 months. Will add the version flags and please send any feedback or requests, trying to stay more on top of this project momentum.
